clang/lib/CodeGen/CodeGenFunction.cpp

```diff
@@ -923,7 +923,8 @@ void CodeGenFunction::StartFunction(GlobalDecl GD, QualType RetTy,
   if (getLangOpts().SYCLIsHost && D && D->hasAttr<SYCLKernelAttr>())
     Fn->addFnAttr("sycl_kernel");
 
-  if (getLangOpts().OpenCL || getLangOpts().SYCLIsDevice) {
+  if (getLangOpts().OpenCL || getLangOpts().SYCLIsDevice ||
+      getLangOpts().SYCLIsHost) {
     // Add metadata for a kernel function.
```

Comment:
We have now enabled calling EmitOpenCLKernelMetadata for the host; this makes all attributes previously applicable only on device also applicable on host.  I don't think that routine has a device/host check, and I am not sure it should either.  But other attributes will now start getting applied on host as well.

Reply:
> We have now enabled calling EmitOpenCLKernelMetadata for the host; this makes all attributes previously applicable only on device also applicable on host. I don't think that routine has a device/host check, and I am not sure it should either. But other attributes will now start getting applied on host as well.

@premanandrao, Yes, this routine does not have a device/host check. Since other attributes do not need to apply on host as well, i decided to add a separate routine to check subgroup attribute on host as well as device.  have uploaded new patch on this. What is your opinion on it. It suplicates some of the codes from EmitOpenCLKernelMetadata routine.